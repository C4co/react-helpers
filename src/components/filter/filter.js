import React, { Fragment } from "react"
import PropTypes from "prop-types"

export function Filter(props) {
  const filtered = props.items.filter((item, index, arr) =>
    props.if(item, index, arr)
  )

  return (
    <Fragment>
      {filtered.map((item, index, arr) => {
        return (
          <Fragment key={index}>
            {props.children(item, index, arr)}
          </Fragment>
        )
      })}
    </Fragment>
  )
}

Filter.propTypes = {
  /** Array of items */
  items: PropTypes.array,

  /** Filter condition */
  if: PropTypes.func,

  /** children component function(item, index, arr) */
  children: PropTypes.func
}
