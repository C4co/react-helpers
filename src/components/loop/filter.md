## Filter

Iteration over multiple itens and filter items based in a condition.

#### Usage
```jsx
  import { Filter } from "react-utl"
```

#### Example:

```jsx
  <Filter items={[1, 2, 3]} if={item => item > 1}>
    {(item, index) => (
      <section>
        <h1> Filtered value => {index} - {item} </h1>
      </section>
    )}
  </Filter>
```

#### Properties:

| Name | Type | Default |
| --- | --- | --- |
| @items | array | [] |
| @children | function(item, index) | - |
| @if | function(item) | - |
