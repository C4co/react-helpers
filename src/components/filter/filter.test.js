import { mount } from "enzyme"
import { Filter } from "./filter"
import React from "react"

describe("Filter component", () => {
  test("Filter numbers", () => {
    const filter = mount(
      <Filter items={[1, 2, 3]} if={item => item > 1}>
        {(item, index) => (
          <h1>
            {index}-{item}
          </h1>
        )}
      </Filter>
    )

    expect(filter.contains(<h1>0-2</h1>)).toBe(true)
    expect(filter.contains(<h1>1-3</h1>)).toBe(true)
    expect(filter).toMatchSnapshot()
  })

  test("Filter strings", () => {
    const result = mount(
      <Filter items={["foo", "fizz", "fozz"]} if={item => item.includes("fo")}>
        {item => (
          <h1>{item}</h1>
        )}
      </Filter>
    )

    expect(result.contains(<h1>foo</h1>)).toBe(true)
    expect(result.contains(<h1>fozz</h1>)).toBe(true)
    expect(result).toMatchSnapshot()
  })
})
