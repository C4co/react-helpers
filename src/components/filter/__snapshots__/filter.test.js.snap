// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component Filter numbers 1`] = `
<Filter
  if={[Function]}
  items={
    Array [
      1,
      2,
      3,
    ]
  }
>
  <h1>
    0
    -
    2
  </h1>
  <h1>
    1
    -
    3
  </h1>
</Filter>
`;

exports[`Filter component Filter strings 1`] = `
<Filter
  if={[Function]}
  items={
    Array [
      "foo",
      "fizz",
      "fozz",
    ]
  }
>
  <h1>
    foo
  </h1>
  <h1>
    fozz
  </h1>
</Filter>
`;
